Document/0x05b-Basic-Security_Testing.md

```diff
@@ -67,225 +94,233 @@ You can get familiar with Magisk reading the official [documentation on GitHub](
+The official Drozer module repository is hosted alongside the main project on GitHub. This is automatically set up in your copy of Drozer. You can search for modules with the `module` command:
 
-The following procedure, which works on the Android emulator that ships with Android Studio 3.x, is for setting up an HTTP proxy on the emulator:
+```shell
+dz> module search tool
+kernelerror.tools.misc.installcert
+metall0id.tools.setup.nmap
+mwrlabs.tools.setup.sqlite3
+```
 
-1. Set up your proxy to listen on localhost and for example port 8080.
-2. Configure the HTTP proxy in the emulator settings:
+For more information about a module, pass the `–d` option to view the module's description:
 
-    - Click on the three dots in the emulator menu bar
-    - Open the Settings Menu
-    - Click on the Proxy tab
-    - Select "Manual proxy configuration"
-    - Enter "127.0.0.1" in the "Host Name" field and your proxy port in the "Port number" field (e.g., "8080")
-    - Tap "Apply"
+```shell
+dz> module  search url -d
+mwrlabs.urls
+    Finds URLs with the HTTP or HTTPS schemes by searching the strings
+    inside APK files.
 
-<img width=600px src="Images/Chapters/0x05b/emulator-proxy.png"/>
+        You can, for instance, use this for finding API servers, C&C
+    servers within malicious APKs and checking for presence of advertising
+    networks.
 
-HTTP and HTTPS requests should now be routed over the proxy on the host machine. If not, try toggling airplane mode off and on.
+```
 
-A proxy for an AVD can also be configured on the command line by using the [emulator command](https://developer.android.com/studio/run/emulator-commandline "Emulator Command") when starting an AVD. The following example starts the AVD Nexus_5X_API_23 and setting a proxy to 127.0.0.1 and port 8080.
+###### Installing Modules
+
+You can install modules with the `module` command:
 
 ```shell
-$ emulator @Nexus_5X_API_23 -http-proxy 127.0.0.1:8080
+dz> module install mwrlabs.tools.setup.sqlite3
+Processing mwrlabs.tools.setup.sqlite3... Already Installed.
+Successfully installed 1 modules, 0 already installed
 ```
 
-##### Installing a CA Certificate on the Virtual Device
+This will install any module that matches your query. Newly installed modules are dynamically loaded into the console and are available immediately.
 
-An easy way to install a CA certificate is to push the certificate to the device and add it to the certificate store via Security Settings. For example, you can install the PortSwigger (Burp) CA certificate as follows:
+### Basic Testing Operations
 
-1. Start Burp and use a web browser on the host to navigate to burp/, then download `cacert.der` by clicking the "CA Certificate" button.
-2. Change the file extension from `.der` to `.cer`.
-3. Push the file to the emulator:
+#### Accessing the Device Shell
 
-    ```shell
-    $ adb push cacert.cer /sdcard/
-    ```
+##### On-device Shell App
+TODO (termux?)
```

Comment:
We could describe it for the sake of completeness, but should also mention that usually using a on-device shell is very tedious compared to a remote shell. But might be good for debugging in case of for example network issues

Reply:
yes, I think the same, also for iOS is useful some times